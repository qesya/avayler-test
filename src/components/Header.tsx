import React from 'react'

interface IHeader {
  title: string;
}

const Header: React.FC<IHeader> = ({ title }) => {
  return (
    <nav className="fixed w-full z-30 py-6 bg-black px-10 md:px-0">
      <div className="container mx-auto flex flex-row justify-between items-center">
        <div>
          <h4 className="font-Rajdhani text-white text-2xl" data-testid="Test-Header">{title}</h4>
        </div>
        <div className="flex flex-row justify-center items-center">
        </div>
      </div>
    </nav>
  )
}

export default Header;
