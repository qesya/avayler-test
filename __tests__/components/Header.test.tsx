import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Header from '../../src/components/Header';

describe('Component:Header', () => {
  it('renders without crashing', async () => {
    const component = render(<Header title="SpaceX" />);
    expect(component).toMatchSnapshot();
  });

  it('it renders the correct title', async () => {
    const expected = "SpaceX";
    const {getByTestId} = render(<Header title="SpaceX" />);
    expect(getByTestId('Test-Header').textContent).toContain(expected);
  });
});