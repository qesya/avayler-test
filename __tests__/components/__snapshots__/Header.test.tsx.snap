// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Header renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="fixed w-full z-30 py-6 bg-black px-10 md:px-0"
      >
        <div
          class="container mx-auto flex flex-row justify-between items-center"
        >
          <div>
            <h4
              class="font-Rajdhani text-white text-2xl"
              data-testid="Test-Header"
            >
              SpaceX
            </h4>
          </div>
          <div
            class="flex flex-row justify-center items-center"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="fixed w-full z-30 py-6 bg-black px-10 md:px-0"
    >
      <div
        class="container mx-auto flex flex-row justify-between items-center"
      >
        <div>
          <h4
            class="font-Rajdhani text-white text-2xl"
            data-testid="Test-Header"
          >
            SpaceX
          </h4>
        </div>
        <div
          class="flex flex-row justify-center items-center"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
